wallet/src/bin/random-wallet-in-mem.rs

```diff
@@ -218,12 +323,53 @@ async fn main() {
     let universal_param = universal_setup_for_test(2usize.pow(16), &mut rng).unwrap();
     let tmp_dir = TempDir::new("random_in_mem_test_sender").unwrap();
     tmp_dirs.push(tmp_dir);
-    let (network, mut wallet) = create_backend_and_sender_wallet(
-        &mut rng,
-        &universal_param,
-        tmp_dirs.last().unwrap().path(),
-    )
-    .await;
+    let (network, mut wallet) = if args.demo_connection {
+        tracing_subscriber::fmt().pretty().init();
+        event!(Level::INFO, "Connecting to Demo Environment");
+        let network = get_network_from_args(&args);
+        let mut wallet = connect_to_demo_backend(
+            &network,
+            &universal_param,
+            &mut rng,
+            tmp_dirs.last().unwrap().path(),
+        )
+        .await;
+        let pk = wallet
+            .generate_user_key("sending account".into(), None)
```

Comment:
`generate_user_key` gets called here and in `connect_to_demo_backend`...is that intentional?

Reply:
Nope good catch, looks like the key generated in `connect_to_demo_backend` is not used.